fl4health/clients/fenda_client.py

```diff
@@ -217,6 +230,14 @@ def compute_loss(
             total_loss += self.cos_sim_loss_weight * cos_sim_loss
             additional_losses["cos_sim_loss"] = cos_sim_loss
 
+        if self.mkmmd_loss_weight:
```

Comment:
Another way that the mkmmd loss can be incorporated is actually in the contrastive loss itself. That is, we use it as a replacement for the `cos_sim` measures in the contrastive calculations. It's probably easier to incorporate it into MOON first, as that approach only has two `cos_sim` calculations. We would have different bases for the two positive and negative measures.

Reply:
This comment is no longer relevant. Closing 🙂